pkg/executor/nginx/nginx_executor.go

```diff
@@ -0,0 +1,120 @@
+package nginx
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"time"
+)
+
+//Executor :
+type Executor interface {
+	PrepareForNginxRun(nginxConfigPath string) *exec.Cmd
+	HandleExit(exitCode int, pid int) int
+	StartLogRotate(pid int, timeInMs int64)
+}
+
+type nginxExecutor struct {
+	nginxExitHandler
+	nginxLogRotate
+}
+
+type nginxExitHandler struct {
+}
+
+type nginxLogRotate struct {
+	paths           []string
+	rotateAfterSize int64
+}
+
+//NewNginxExecutor :
+func NewNginxExecutor(rotateAfterSize int64, logfiles []string) Executor {
+	return nginxExecutor{
+		nginxExitHandler{},
+		nginxLogRotate{
+			paths:           logfiles,
+			rotateAfterSize: rotateAfterSize,
+		},
+	}
+}
+
+func (m nginxExecutor) PrepareForNginxRun(nginxConfigPath string) *exec.Cmd {
+	cmd := exec.Command("sh", "-c", fmt.Sprintf("exec nginx -g 'daemon off;' -c %s", nginxConfigPath))
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd
+}
+
+func (m nginxLogRotate) StartLogRotate(pid int, checkRotateAfterMs int64) {
+	ticker := time.NewTicker(time.Duration(checkRotateAfterMs) * time.Millisecond)
+	done := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-done:
+				return
+			case t := <-ticker.C:
+				//Check file size of all files
+				for _, path := range m.paths {
+
+					fileinfo, err := os.Stat(path)
+					if err != nil {
+						if os.IsNotExist(err) {
+							continue
+						}
+						logrus.Errorf("Could not stat file %s. Err %v", path, err)
+						continue
+					}
+
+					size := fileinfo.Size() >> 20
+					//close file
+
+					if size >= m.rotateAfterSize {
+						logrus.Debugf("Rotate log at %s", t)
+						if err := m.rotate(pid, path); err != nil {
+							logrus.Errorf("Could not rotate logfile %s", path)
+						}
+					}
+				}
+			}
+		}
+	}()
+}
+
+func (m nginxLogRotate) rotate(pid int, path string) error {
+
+	var extension = filepath.Ext(path)
+	var base = path[0 : len(path)-len(extension)]
+	oldLog := fmt.Sprintf("%s.1%s", base, extension)
```

Comment:
At du får en teller i navnet impliserer at det kommer "2" etter "1", mens i realiteten blir bare 1 overskrevet når den tid kommer. Det er ikke galt, men jeg ville ha vurdert om det ikke hadde vært bedre å rotert til en suffix `.previous` eller `.old´

Husk å dobbeltsjekke at du får lov til å gjøre mv til en fil som eksisterer (i tilfelle det skulle være noen globbing-regler i effekt).

Reply:
- Formatet må være på formen: *.[1-9].[access|log]. Her er det fluentbit konfigurasjonen som legger føringer. 
- os.Rename sørger for at filen blir overskrevet om denne finnes.